### PR TITLE
#27345 Added Symfony\Component\Lock\Store\MongoDbStore

### PR DESCRIPTION
**Pull request by kralos**
Added Lock MongoDbStore documentation to symfony 5.1

### CODE REVIEW

**Comment by kralos** [comment]
see https://github.com/symfony/symfony/pull/31889

**Comment by kralos** [comment]
`git rebase origin/4.4` and squashed

File: components/lock.rst [hunk 1]
```diff
@@ -275,6 +276,39 @@ support blocking, and expects a TTL to avoid stalled locks::
...
+
+    The ``MongoDbStore`` was introduced in Symfony 4.4.
+
+The MongoDbStore saves locks on a MongoDB server, it requires a
+``\MongoDB\Client`` connection from `mongodb/mongodb`_. This store does not
+support blocking and expects a TTL to avoid stalled locks::
+
+    use Symfony\Component\Lock\Store\MongoDbStore;
+
+    $mongoClient = new \MongoDB\Client('mongo://localhost/');
```
Comment (OskarStark) [inline components/lock.rst]:
```suggestion
    $client = new \MongoDB\Client('mongo://localhost/');
```

File: components/lock.rst [hunk 2]
```diff
@@ -275,6 +276,39 @@ support blocking, and expects a TTL to avoid stalled locks::
...
+
+.. versionadded:: 4.4
+
+    The ``MongoDbStore`` was introduced in Symfony 4.4.
+
+The MongoDbStore saves locks on a MongoDB server, it requires a
+``\MongoDB\Client`` connection from `mongodb/mongodb`_. This store does not
+support blocking and expects a TTL to avoid stalled locks::
+
+    use Symfony\Component\Lock\Store\MongoDbStore;
```
Comment (OskarStark) [inline components/lock.rst]:
```suggestion
    use MongoDB\Client;
    use Symfony\Component\Lock\Store\MongoDbStore;
```

File: components/lock.rst [hunk 3]
```diff
@@ -275,6 +276,39 @@ support blocking, and expects a TTL to avoid stalled locks::
...
+
+    $mongoClient = new \MongoDB\Client('mongo://localhost/');
+
+    $options = [
+        'database' => 'my-app',
+    ];
+
+    $store = new MongoDbStore($mongoClient, $options);
```
Comment (OskarStark) [inline components/lock.rst]:
```suggestion
    $store = new MongoDbStore($client, $options);
```

File: components/lock.rst [hunk 4]
```diff
@@ -275,6 +276,39 @@ support blocking, and expects a TTL to avoid stalled locks::
...
+
+The ``MongoDbStore`` takes the following ``$options``:
```
Comment (OskarStark) [inline components/lock.rst]:
```suggestion
The ``MongoDbStore`` takes the following options:
```

**Comment by kralos** [comment]
Since @fabpot just merged https://github.com/symfony/symfony/pull/31889 I've re-based this from `symfony-docs/master` and bumped the introduced in version number to `5.1`

**Comment by kralos** [comment]
The code for this has been merged into `symfony/symfony` `master`. It will be released in `symfony` `5.1`. I've updated this merge request accordingly

File: components/lock.rst [hunk 5]
```diff
@@ -267,6 +268,67 @@ support blocking, and expects a TTL to avoid stalled locks::
...
+
+MongoDB Connection String:
+
+- ``$options['database']`` is used otherwise ``/path`` from the DSN, at least one is mandatory
+- ``$options['collection']`` is used otherwise ``?collection=`` from the DSN, at least one is mandatory
+
+.. note::
+
+    The ``collection`` querystring parameter is not part of the `MongoDB Connection String`_ definition.
+    It is simply used to allow constructing a ``MongoDbStore`` using a `Data Source Name (DSN)`_ without ``$options``.
```
Comment (OskarStark) [inline components/lock.rst]: Please remove “simply” or replace it by “just”

File: components/lock.rst [hunk 6]
```diff
@@ -546,6 +608,46 @@ method uses the Memcached's ``flush()`` method which purges and removes everythi
...
+MongoDbStore
+~~~~~~~~~~~~
+
+.. caution::
+
+    The locked resource name is indexed in the ``_id`` field of the lock
+    collection. Beware that in MongoDB an indexed field's value can be
+    `a maximum of 1024 bytes in length`_ inclusive of structural overhead.
+
+A TTL index MUST BE used to automatically clean up expired locks.
```
Comment (OskarStark) [inline components/lock.rst]: If bold then **must be**

File: components/lock.rst (hunk 5 above)
Comment (kralos) [reply components/lock.rst]: removed

File: components/lock.rst (hunk 6 above)
Comment (kralos) [reply components/lock.rst]: in the interest of consistency with all other usage of `must be` / `must not be` i've just made it lower case / not bold.

**Review by OskarStark — APPROVED** [review]

**Review by javiereguiluz — APPROVED** [review]
@kralos thanks a lot for your contribution. I tried to merge it with the internal tool we use to merge Symfony PRs ... but I faced an error that I couldn't fix myself. So, could you please rebase your PR one last time? Thanks!

**Comment by kralos** [comment]
@javiereguiluz I've fixed the branch, also noticed the DBAL link was broken (Doctrine dropped the DBAL dir using psr-4) so i've updated it

**Comment by javiereguiluz** [comment]
Merged! Thanks a lot for the extra effort needed to rebase and get this PR ready.

**Pull request merged**